web3/datastructures.py

```diff
@@ -123,6 +123,26 @@ def __eq__(self, other: Any) -> bool:
             return False
 
 
+def tupleize_lists_nested(d: Mapping[TKey, TValue]) -> AttributeDict[TKey, TValue]:
+    """
+    Lists inside dicts will throw an error if attempted to be hashed.
+    This method converts them to tuples, rendering them hashable.
+    """
+
+    def _to_tuple(lst: List[Any]) -> Any:
+        return tuple(_to_tuple(i) if isinstance(i, list) else i for i in lst)
+
+    ret = dict()
+    for k, v in d.items():
+        if isinstance(v, list):
+            ret[k] = _to_tuple(v)
+        elif isinstance(v, dict) or isinstance(v, ReadableAttributeDict):
```

Comment:
I think we may want to be less selective here

```suggestion
        elif isinstance(v, Mapping):
```